selfservice/strategy/oidc/provider.go

```diff
@@ -27,6 +27,7 @@ type Provider interface {
 type OAuth2Provider interface {
 	Provider
 	AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption
+	AccessTokenURLOptions(r *http.Request) []oauth2.AuthCodeOption
```

Comment:
As this method is only relevant for VK, we could also extract it into a separate interface that only the VK provider implements. Then, in `strategy.go` we assert on the interface and add the options if the provider implements it. 

That way, we don't have to give empty implementations to every other provider. WDYT?